enterprise/reporting/error_reporting.go

```diff
@@ -386,6 +393,33 @@ func (edr *ErrorDetailReporter) mainLoop(ctx context.Context, c types.SyncerConf
 	}
 }
 
+func (edr *ErrorDetailReporter) vacuum(ctx context.Context, dbHandle *sql.DB, c types.SyncerConfig) {
+	tags := edr.getTags(c.Label)
+	vacuumDuration := edr.stats.NewTaggedStat(StatReportingVacuumDuration, stats.TimerType, tags)
+
+	vacuumStart := time.Now()
+	var sizeEstimate int64
+	if err := dbHandle.QueryRowContext(
+		ctx,
+		`SELECT pg_table_size(oid) from pg_class where relname = $1`, ErrorDetailReportsTable,
+	).Scan(&sizeEstimate); err != nil {
+		edr.log.Errorn(
+			fmt.Sprintf(`Error getting %s table size estimate`, ErrorDetailReportsTable),
+			logger.NewErrorField(err),
+		)
+	}
+	if sizeEstimate > config.GetInt64("Reporting.errorReporting.vacuumThresholdBytes", 5*bytesize.GB) {
+		vaccumStatement := fmt.Sprintf("vacuum full analyze %s", pq.QuoteIdentifier(ErrorDetailReportsTable))
+		if _, err := dbHandle.ExecContext(ctx, vaccumStatement); err != nil {
+			edr.log.Errorn(
+				fmt.Sprintf(`Error vacuuming %s table`, ErrorDetailReportsTable),
+				logger.NewErrorField(err),
+			)
+		}
```

Comment:
We can probably move the stat and the duration during running the query.
```suggestion
        vacuumStart := time.Now()
        vacuumDuration := edr.stats.NewTaggedStat(StatReportingVacuumDuration, stats.TimerType, tags)
		vaccumStatement := fmt.Sprintf("vacuum full analyze %s", pq.QuoteIdentifier(ErrorDetailReportsTable))
		if _, err := dbHandle.ExecContext(ctx, vaccumStatement); err != nil {
			edr.log.Errorn(
				fmt.Sprintf(`Error vacuuming %s table`, ErrorDetailReportsTable),
				logger.NewErrorField(err),
			)
		}
		vacuumDuration.Since(vacuumStart)	
```